### PR TITLE
Add wattle, daubed, plastered, and timber-framed walls

### PR DESCRIPTION
**Pull request by noelle-lavenza**
Adds wattle, daubed, plastered, and timber-framed walls. Sprites by Doe.
Fixes strings that misspell `separate` as `seperate`.
Updates the cabinet sprites since I forgot to commit that at some point.

There's currently no way to plaster walls. I should probably also make it so you can't plaster (or recolor) them using a paint sprayer? I also haven't tested the daubing part. And I haven't added a crafting recipe for wattle walls. Uhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh maybe I should draft this PR

### CODE REVIEW

File: code/game/turfs/walls/wall_wattle.dm [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+/turf/wall/wattle/Initialize(ml, materialtype, rmaterialtype)
+	if(ispath(daubing_material))
+		daubing_material = GET_DECL(daubing_material)
+	return ..()
+
+// Daubing with clay or soil
+/turf/wall/wattle/attackby(obj/item/W, mob/user, click_params)
+	if(isnull(daubing_material))
+		var/static/list/daub_materials = list( // Does not include subtypes.
```
Comment (MistakeNot4892) [inline code/game/turfs/walls/wall_wattle.dm:26]: Dirt material could be used to represent adobe?

Comment (MistakeNot4892) [reply code/game/turfs/walls/wall_wattle.dm:26]: Nvm that's soil.

File: code/unit_tests/turf_icons.dm [hunk 2]
```diff
@@ -140,6 +140,9 @@
 /turf/wall/natural/get_turf_validation_corner_states()
 	return list("", "shine")
 
+/turf/wall/log/get_turf_validation_corner_states()
```
Comment (MistakeNot4892) [inline code/unit_tests/turf_icons.dm:143]: Does `brick` need one too?

Comment (noelle-lavenza) [reply code/unit_tests/turf_icons.dm:143]: No LOL it didn't fail CI, they have paint/etc states

File: code/game/turfs/walls/wall_icon.dm [hunk 3]
```diff
@@ -54,6 +58,14 @@
 /turf/wall/proc/apply_reinf_overlay()
 	. = istype(reinf_material)
 
+/// Gets the base wall colour for icon rendering. Can be overridden on wall subtypes. Not equivalent to get_color().
+/turf/wall/proc/get_base_color()
+	return material.color
```
Comment (MistakeNot4892) [inline code/game/turfs/walls/wall_icon.dm:64]: Might need an istype() check on material depending on how early you can expect this to get called.

File: code/game/turfs/walls/wall_icon.dm [hunk 4]
```diff
@@ -54,6 +58,14 @@
...
+
+/// Gets the reinforcement colour. Can be overridden so that some wall types don't apply paint colour to their reinforcements.
+/turf/wall/proc/get_reinf_color()
+	return paint_color ? paint_color : reinf_material?.color
```
Comment (MistakeNot4892) [inline code/game/turfs/walls/wall_icon.dm]: `paint_color || reinf_material?.color` ?

File: code/game/turfs/walls/wall_icon.dm (hunk 3 above)
Comment (MistakeNot4892) [reply code/game/turfs/walls/wall_icon.dm:64]: Looks like it's in icon code so would be unsurprised if this runtimes due to material being a type on some neighbor that gets poked during turf init.

File: code/game/turfs/walls/wall_icon.dm [hunk 5]
```diff
@@ -124,17 +136,17 @@
 
 	if(apply_reinf_overlay())
```
Comment (MistakeNot4892) [inline code/game/turfs/walls/wall_icon.dm:138]: Null check on `reinf_icon` here possibly?

Comment (MistakeNot4892) [reply code/game/turfs/walls/wall_icon.dm:138]: Or in `apply_reinf_overlay()`

File: code/unit_tests/turf_icons.dm [hunk 6]
```diff
@@ -127,15 +127,22 @@
...
-	if(paint_color)
+	. = list("")
+	if(!material)
+		CRASH("[type] lacks a material!")
+	if(material?.wall_flags & WALL_HAS_EDGES)
+		. |= "other"
+	if(paint_color || material?.wall_flags & PAINT_PAINTABLE)
 		. |= "paint"
-	if(stripe_color)
+	if(stripe_color || material?.wall_flags & PAINT_STRIPABLE)
```
Comment (MistakeNot4892) [inline code/unit_tests/turf_icons.dm]: Parens

File: code/game/turfs/walls/wall_icon.dm (hunk 3 above)
Comment (noelle-lavenza) [reply code/game/turfs/walls/wall_icon.dm:64]:
```
	if(!istype(material))
		return

	refresh_connections()
	update_wall_icon()
```
update_wall_icon never runs if material is not a datum

File: code/game/turfs/walls/wall_icon.dm (hunk 5 above)
Comment (noelle-lavenza) [reply code/game/turfs/walls/wall_icon.dm:138]: reinf_icon must always be set if reinf_material is set and use_reinf_state is also set—actually i guess the material could just not have one.

File: code/game/turfs/walls/wall_icon.dm (hunk 3 above)
Comment (noelle-lavenza) [reply code/game/turfs/walls/wall_icon.dm:64]: i added a comment to make that clear

**Comment by noelle-lavenza** [comment]
> There's currently no way to plaster walls. I should probably also make it so you can't plaster (or recolor) them using a paint sprayer? I also haven't tested the daubing part. And I haven't added a crafting recipe for wattle walls. Uhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh maybe I should draft this PR

i am going to say:
- the paint sprayer can do whatever for now, i'm not up for making a generalisable solution to that right now. it's an edge case of an edge case
- daubing works hooray
- i am honestly okay with it being mapping only for now but could also just make it crafted with wood rods

basically: ready for review

File: code/game/turfs/walls/wall_icon.dm (hunk 5 above)
Comment (MistakeNot4892) [reply code/game/turfs/walls/wall_icon.dm:138]: That is my concern, yeah.

Comment (MistakeNot4892) [reply code/game/turfs/walls/wall_icon.dm:138]: Oh you added it to the proc, easy.

**Review by MistakeNot4892 — APPROVED** [review]
I think the implementation needs some refinement but we're going to need better systems for variable turfs like this to make that worthwhile. Fine with this going in as-is.

**Pull request merged**